# U.S. Medical Insurance Costs

- Find out the average age of the patients in the dataset.
- Analyze where a majority of the individuals are from.
- Look at the different costs between smokers vs. non-smokers.
- Figure out what the average age is for someone who has at least one child in this dataset.

In [95]:
import csv
import pandas as pd
import numpy as np

In [96]:
# with open('insurance.csv') as insurance:
#     print(insurance.read())

age = []
sex = []
bmi = []
children = []
smoker = []
region = []
charges = []

with open('insurance.csv') as insurance_file:
    reader = csv.DictReader(insurance_file)
    for row in reader:
        age.append(row['age'])
        sex.append(row['sex'])
        bmi.append(row['bmi'])
        children.append(row['children'])
        smoker.append(row['smoker'])
        region.append(row['region'])
        charges.append(row['charges'])

- Find out the average age of the patients in the dataset.

In [97]:
def avg_age(age_list):
    total_age = 0
    for i in age_list:
        total_age += int(i)
    
    average_age = round(total_age / len(age_list))
    return average_age

avg_age(age)

39

the average age is 39 years old

- Analyze where a majority of the individuals are from.

In [98]:
def region_details(region_list):
    unique_regions = []
    region_count = []
    for i in region_list:
        if i in unique_regions:
            continue
        else:
            unique_regions.append(i)
    
    for i in unique_regions:
        region_count.append(region_list.count(i))
    
    region_summary = {region:count for region, count in zip(unique_regions, region_count)}
    return region_summary

region_details(region)

{'southwest': 325, 'southeast': 364, 'northwest': 325, 'northeast': 324}

the majority of the people are from the south, with the greatest number from the southeast

- Look at the different costs between smokers vs. non-smokers.

In [99]:
def avg_smoke_status_cost(smoker, charges):
    smoker_total = 0
    smoker_count = 0
    nonsmoker_total = 0
    nonsmoker_count = 0
    cost_by_status = {}
    for i in range(len(smoker)):
        status = smoker[i]
        if status == 'yes':
            smoker_count += 1
            smoker_total += float(charges[i])
        else:
            nonsmoker_count += 1
            nonsmoker_total += float(charges[i])
    smoker_avg = round(smoker_total/smoker_count,2)
    nonsmoker_avg = round(nonsmoker_total/nonsmoker_count,2)
    cost_by_status['smoker'] = smoker_avg
    cost_by_status['non-smoker'] = nonsmoker_avg
    return cost_by_status

avg_smoke_status_cost(smoker, charges)

{'smoker': 32050.23, 'non-smoker': 8434.27}

the average charges for smokers is 32050.23 and the average charges for non-smokers is 8434.27

- Figure out what the average age is for someone who has at least one child in this dataset.

In [100]:
def avg_age_parent(children, age):
    total_age = 0
    for i in range(len(children)):
        child_count = int(children[i])
        if child_count > 0:
            total_age += int(age[i])
        avg_age = round(total_age/len(age))
    return avg_age

avg_age_parent(children, age)

23

the average age of someone with at least one child is 23